lib/hanami/action/rack/file.rb

```diff
@@ -15,15 +15,15 @@ class File
         # @since 0.4.3
         # @api private
         def initialize(path)
-          @file = ::Rack::File.new(nil)
+          @file = ::Rack::File.new(Dir.pwd)
           @path = path
         end
 
         # @since 0.4.3
         # @api private
         def call(env)
-          @file.path = @path.to_s
-          @file.serving(env)
+          env["PATH_INFO"] = @path
```

Comment:
And you can see that now Rack take path from `path_info` env variable ([link](https://github.com/rack/rack/blob/master/lib/rack/file.rb#L40))